app.py

```diff
@@ -34,30 +35,38 @@ def download():
 
         if not url:
             return jsonify({"error": "URL is required"}), 400
-        if not extension:
-            extension = "mp4"
-        if extension == "mp3" and not audio_only:
-            return jsonify({"status": "error", "Wrong extension or missing parameters": f"{extension} extension "
-                                                                                        f"requires audio_only to be "
-                                                                                        f"True"}), 500
-
-        yt = YouTube(url)
-        # Append v or a respectively if its an audio or video file
-        if audio_only:
-            video_stream = yt.streams.filter(
-                only_audio=audio_only, file_extension="mp4").first()
-            video_filename = f'a-{video_stream.default_filename.split(".")[0]}.{extension}'
-        else:
-            video_stream = yt.streams.filter(progressive=True,
-                                             file_extension=extension).first()
-            video_filename = f'v-{video_stream.default_filename.split(".")[0]}.{extension}'
-
-        if check_cache(video_filename):
-            return send_from_directory(DOWNLOAD_FOLDER, video_filename)
-        else:
-            video_stream.download(
-                filename=f'{DOWNLOAD_FOLDER}/{video_filename}')
-            return send_from_directory(DOWNLOAD_FOLDER, video_filename)
+        if "instagram" in url:
+            shortcode = url.split('p/')[1].strip('/ ')
+            L = instaloader.Instaloader(dirname_pattern= DOWNLOAD_FOLDER + '/' + shortcode )
+            post = instaloader.Post.from_shortcode(L.context, shortcode)
+            L.download_post(post,shortcode)
+            dir = os.listdir(path=DOWNLOAD_FOLDER + '/' + shortcode)
+            return send_from_directory(DOWNLOAD_FOLDER + '/' + shortcode, dir[2])
```

Comment:
Instead of using dir[2], use pathlib module's Path.suffix attribute, see following line:
`pathlib.Path("downloads/CyfQDqZN909/2023-10-17_05-33-01_UTC.jpg").suffix`

![image](https://github.com/DawiAlotaibi/MediaDownloader-API/assets/76831852/0dd892ab-b1e1-4d18-987a-8b35f5c45abe)

Reply:
Please Check my last commit.